python/lsst/daf/butler/tests/testFormatters.py

```diff
@@ -45,10 +46,19 @@ def write(self, inMemoryDataset: Any) -> str:
 class FormatterTest(Formatter):
     """A test formatter that does not need to format anything."""
 
-    supportedWriteParameters = frozenset({"min", "max", "median", "comment", "extra"})
+    supportedWriteParameters = frozenset({"min", "max", "median", "comment", "extra", "recipe"})
 
     def read(self, component: Optional[str] = None) -> Any:
         raise NotImplementedError("Type does not support reading")
 
     def write(self, inMemoryDataset: Any) -> str:
         raise NotImplementedError("Type does not support writing")
+
+    @staticmethod
```

Comment:
I've come to hate `@staticmethod`, as it prevents you from using `super()`. I suggest using `@classmethod`: even if you're not using the class, a subclass may want to override and use `super()`.